airbyte-scheduler/client/src/main/java/io/airbyte/scheduler/client/DefaultSynchronousSchedulerClient.java

```diff
@@ -177,4 +189,34 @@ private <T> void track(final UUID jobId,
 
   }
 
+  private <S, T> void reportError(final ConfigType configType,
+                                  final S jobConfig,
+                                  final T jobOutput,
+                                  final UUID connectorDefinitionId,
+                                  final UUID workspaceId) {
+
+    switch (configType) {
+      case CHECK_CONNECTION_SOURCE -> jobErrorReporter.reportSourceCheckJobFailure(
+          connectorDefinitionId,
+          workspaceId,
+          ((ConnectorJobOutput) jobOutput).getFailureReason(),
+          (JobCheckConnectionConfig) jobConfig);
+      case CHECK_CONNECTION_DESTINATION -> jobErrorReporter.reportDestinationCheckJobFailure(
+          connectorDefinitionId,
+          workspaceId,
+          ((ConnectorJobOutput) jobOutput).getFailureReason(),
+          (JobCheckConnectionConfig) jobConfig);
+      case DISCOVER_SCHEMA -> jobErrorReporter.reportDiscoverJobFailure(
+          connectorDefinitionId,
+          workspaceId,
+          ((ConnectorJobOutput) jobOutput).getFailureReason(),
+          (JobDiscoverCatalogConfig) jobConfig);
+      case GET_SPEC -> jobErrorReporter.reportSpecJobFailure(
+          ((ConnectorJobOutput) jobOutput).getFailureReason(),
+          (JobGetSpecConfig) jobConfig);
+      default -> throw new IllegalArgumentException(
```

Comment:
since this is for reporting, should we instead log rather than throw an exceptoin?

Reply:
👍🏻 done in [da903d4](https://github.com/airbytehq/airbyte/pull/14818/commits/da903d43e09f4e97e74a441d6eedf394de0532f2)